src/server/ua_server_binary.c

```diff
@@ -519,8 +519,9 @@ processMSG(UA_Server *server, UA_SecureChannel *channel,
         session = &anonymousSession;
     }
 
-    /* Trying to use a non-activated session? */
-    if(sessionRequired && !session->activated) {
+    /* Trying to use a non-activated session?
+     * Do not allow if request is of type CloseSessionRequest */
+    if(sessionRequired && !session->activated && strcmp(requestType->typeName, "CloseSessionRequest")) {
```

Comment:
I'm not sure this has the intended effect.
This allows access if
- The service requires a session
- The session is not activated
- The service is **not** a CloseSessionRequest
Please close this pontential security hole.

Can you move the test for session->activated here?
https://github.com/open62541/open62541/blob/4209f74e2176f75f67b4f80962909c69cdb54126/src/server/ua_services_session.c#L368